internal/pkg/cli/deploy/workload.go

```diff
@@ -401,6 +401,7 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 	if len(buildArgsPerContainer) == 0 {
 		return nil
 	}
+	isMultipleContainerImages := len(buildArgsPerContainer) > 1
```

Comment:
Does it make more sense for us to call a separate function if `isMultipleContainerImages` is true (after `repository.Login()`)?